### PR TITLE
[Merged by Bors] - feat: `DecidableEq` for `Multiset.ToType`

### PR DESCRIPTION
**Pull request by kmill**
Also clean up some of the documentation, for instance removing the claim that you shouldn't use `Multiset.ToType` directly.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by madvorak** [comment]
LGTM

File: Mathlib/Data/Multiset/Fintype.lean [hunk 1]
```diff
@@ -37,19 +39,17 @@ open BigOperators
...
 
-/-- Auxiliary definition for the `hasCoeToSort` instance. This prevents the `hasCoe m α`
-instance from inadvertently applying to other sigma types. One should not use this definition
-directly. -/
--- Porting note: @[nolint has_nonempty_instance]
-def Multiset.ToType (m : Multiset α) : Type _ :=
-  Σx : α, Fin (m.count x)
+/-- Auxiliary definition for the `CoeSort` instance. This prevents the `CoeSort m α`
+instance from inadvertently applying to other sigma types. -/
+def Multiset.ToType (m : Multiset α) : Type _ := (x : α) × Fin (m.count x)
```
Comment (madvorak) [inline Mathlib/Data/Multiset/Fintype.lean:44]: Why don't you write it with \Sigma anymore? I found it more readable as it was.

File: Mathlib/Data/Multiset/Fintype.lean [hunk 2]
```diff
@@ -37,19 +39,17 @@ open BigOperators
...
 variable {α : Type*} [DecidableEq α] {m : Multiset α}
 
-/-- Auxiliary definition for the `hasCoeToSort` instance. This prevents the `hasCoe m α`
-instance from inadvertently applying to other sigma types. One should not use this definition
-directly. -/
--- Porting note: @[nolint has_nonempty_instance]
-def Multiset.ToType (m : Multiset α) : Type _ :=
-  Σx : α, Fin (m.count x)
+/-- Auxiliary definition for the `CoeSort` instance. This prevents the `CoeSort m α`
+instance from inadvertently applying to other sigma types. -/
```
Comment (eric-wieser) [inline Mathlib/Data/Multiset/Fintype.lean:43]:
```suggestion
/-- Auxiliary definition for the `CoeSort` instance. This prevents the `CoeOut m α`
instance from inadvertently applying to other sigma types. -/
```

File: Mathlib/Data/Multiset/Fintype.lean (hunk 1 above)
Comment (kmill) [reply Mathlib/Data/Multiset/Fintype.lean:44]: I like dependent product notation more. Readability is subjective -- I think it's more readable than \Sigma.

**Review by eric-wieser — APPROVED** [review]
bors merge

Thanks!

**Pull request closed without merging**